manta-js/package/src/api/index.js

```diff
@@ -117,26 +121,42 @@ export default class Api {
       await this.api.isReady;
 
       this._log('checkpoint ' + JSON.stringify(checkpoint));
-      let result = await this.api.rpc.mantaPay.pull_ledger_diff(
+      let result = await this.api.rpc.mantaPay.dense_pull_ledger_diff(
         checkpoint,
         this.maxReceiversPullSize,
         this.maxSendersPullSize
       );
 
       this._log('pull result ' + JSON.stringify(result));
 
-      const receivers = result.receivers.map((receiver_raw) => {
-        return [
-          this._utxo_to_json(receiver_raw[0]),
-          this._full_incoming_note_to_jons(receiver_raw[1])
-        ];
-      });
-      const senders = result.senders.map((sender_raw) => {
-        return [
-          Array.from(sender_raw[0].toU8a()),
-          this._outgoing_note_to_json(sender_raw[1]),
-        ];
-      });
+      const decodedReceivers = $Receivers.decode(
+        base64Decode(result.receivers.toString())
+      );
+      $.assert($Receivers, decodedReceivers);
+      let receivers = [];
+      if (decodedReceivers.length > 0) {
```

Comment:
is this `if` condition needed? I thought that if decodedReceivers length is zero, inner map operation will not executed, thus result will be empty array.

Reply:
Yeah, it's the code for debugging, it caused wasm layer crash. Now it has been fixed.